### ISSUE TITLE
java.lang.IllegalArgumentException: Can't define method not in interfaces: with_meta, compiling:(clj_http/headers.clj:105:1)

### ISSUE THREAD

**Issue by xperian** [dakrone/clj-http#254]
latest build from source throws an exception.

```
[org.clojure/clojure "1.7.0-alpha5"]
```

http://pastebin.com/1Ld4D8Q7

**Comment by joegallo** [comment]
I'm not sure what you're seeing exactly, a brand new checkout seems to work just fine for me. 

https://gist.github.com/joegallo/8ef50a70ada3bd5781fa

What version on leiningen are you using?

**Comment by xperian** [comment]
```
Leiningen 2.5.1 on Java 1.8.0_25 Java HotSpot(TM) 64-Bit Server VM
```
your Clojure version is 1.6.0 . Could you try again with  org.clojure/clojure "1.7.0-alpha5" ?

**Comment by joegallo** [comment]
Sure thing. https://gist.github.com/joegallo/8bdf2e4637c2ea37e44f

**Comment by xperian** [comment]
thanks for showing this.
In my case I had many other dependencies. I dont have much time to dive into this problem atm ,so I rolled back to previous cleaner deps version. clj-http is working , with Clojure 1.7. I will look into this problem later .

**Comment by joegallo** [comment]
My best guess is that something you're depending on is bringing in an older version of potemkin, probably.

**Comment by xperian** [comment]
I checked this (I saw one issue from Ztellman). I made sure that Potmkin was the latest,but it did not work. I will check into this later. For now everything good and I have to other important stuff.

**Comment by dakrone** [comment]
@xperian are you using any custom plugins or deps in `~/.lein/profiles.clj` that might be interfering? Can you post your `~/.lein/profiles.clj`?

**Comment by xperian** [comment]
I found the reason. I imported leiningen.
```
[clj-stacktrace "0.2.8" :exclude [org.clojure/clojure]]
                 [spyscope "0.1.4" :exclude [org.clojure/clojure clj-time]]
                 [leiningen "2.5.1" :exclude [clj-http]]
                 [io.aviso/pretty "0.1.8" :exclude [clj-http]]
                 [im.chit/vinyasa "0.3.3"  :exclude [clj-http]]
```


```
lein repl
WARNING: update already refers to: #'clojure.core/update in namespace: clj-http.client, being replaced by: #'clj-http.client/update
2015-Mar-02 18:44:27 +0800 DataScientist ERROR [scholar.service.seg] - Attempting to call unbound fn: #'clj-http.client/update
2015-Mar-02 18:44:27 +0800 DataScientist ERROR [scholar.service.moses] - Attempting to call unbound fn: #'clj-http.client/update
nREPL server started on port 53022 on host 127.0.0.1 - nrepl://127.0.0.1:53022
REPL-y 0.3.5, nREPL 0.2.6
Clojure 1.7.0-alpha5
Java HotSpot(TM) 64-Bit Server VM 1.8.0_25-b17
    Docs: (doc function-name-here)
          (find-doc "part-of-name-here")
  Source: (source function-name-here)
 Javadoc: (javadoc java-object-or-class-here)
    Exit: Control+D or (exit) or (quit)
 Results: Stored in vars *1, *2, *3, an exception in *e

scholar.system=> exit
Bye for now!
```

**Comment by xperian** [comment]
I configured my project as this one.
https://github.com/zcaudate/vinyasa

I placed clj-http dep at top the top of my project file.

**Comment by dakrone** [comment]
@xperian I'm unclear about this, is it still a problem? Was it fixed by not importing leiningen?

When I try using a REPL with the vinyasa project it seems to work okay.

**Comment by xperian** [comment]
After deleting leiningen ,all is well,so I am closing this. I dont have nor time or enough background to dig deeper for now .  maybe months later, I will be  ready to dig. thanks all.

**Comment by emidln** [comment]
This is actually a problem with newer versions of potemkin. potemkin.collections.PotemkinMap now requires meta and with-meta to be defined in implementations (as of 0.3.11) although this was unfortunately not documented until after 0.3.11 was released (it's in 0.3.12).
